orc/src/main/java/org/apache/iceberg/orc/OrcRowWriter.java

```diff
@@ -35,4 +37,9 @@
    * @throws IOException if there's any IO error while writing the data value.
    */
   void write(T row, VectorizedRowBatch output) throws IOException;
+
+  /**
+   * Returns a stream of {@link FieldMetrics} that this OrcRowWriter keeps track of.
+   */
+  Stream<FieldMetrics> metrics();
```

Comment:
why some method signatures of `metrics` have default, but some others below do not?

Reply:
Currently value writers all have a default, and this is because we are only tracking this metrics for float and wrapper types, declaring a default will save other types from declaring empty stream. This row writer on the other hand handles row writing, and it will always need to read from value writers' metrics, and thus default wouldn't help much on this aspect. Although I guess we want to prevent breaking people's code if they implement their own version of row writer, that I'll add a default here to avoid that.